### ISSUE TITLE
Do no show error for type_check_only class inheritance in TYPE_CHECKING block

### ISSUE THREAD

**Issue by Roang-zero1** [microsoft/pyright#10956]
**Describe the bug**

When using `type_check_only` in a `TYPE_CHECKING` block pyright warns `"Class"  is marked as @type_check_only and can be used only in type annotations`.
When using the same construct in a stubs file there is no similar warning

**Code or Screenshots**
`main.py`
```python
from typing import TYPE_CHECKING

if TYPE_CHECKING:
    from typing import type_check_only

    @type_check_only
    class SomeClass():
        pass
    
    @type_check_only
    class Another(SomeClass): # <-- "SomeClass" is marked as @type_check_only and can be used only in type annotations
        pass
```
`main.pyi`
```python
from typing import type_check_only

@type_check_only
class SomeClass():
    pass

@type_check_only
class Another(SomeClass):
    pass
```

**VS Code extension or command-line**
Tested with:
* VSCode with `Pylance 2025.8.2` with `pyright 1.1.394` in the venv
* `pyright 1.1.405` via CLI in the reproduction repo

**Comment by erictraut** [comment]
Pyright is working as designed here, so this isn't a bug.

As a principle, pyright never special-cases the handling of code within a `TYPE_CHECKING` block. The `TYPE_CHECKING` symbol is intended as a way to "lie" to the type checker about what it should see. There are many reasons it is used, and IMO it would be dangerous for a type checker to make assumptions about why it's being used.

**Comment by Roang-zero1** [comment]
Ok, thanks for the info. 
So the correct way would be to put the definitions for these types into stub files? (As in the working example)
